cmd/honeyalb/main.go

```diff
@@ -147,7 +147,7 @@ For reference see this link:
 
 http://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html#enable-access-logging
 `, lbName)
-					os.Exit(1)
+					continue
```

Comment:
I'd prefer if you implemented it like this: https://github.com/quintoandar/honeyaws/commit/060c69ba5bb9841a627c764637c88d05b885cd04

i.e. _you only continue if there is no args supplied_, like, if you call the binary like

`honeyalb`

Because then it will try to pull data from _all_ alb/nlb.

With your solution, if I call a single (or a list) of alb, and they're _all_ unavailable, the binary will keep running, but I'll get no logs/no error message.

Reply:
Error messages you will get a lot of them. There's a line above that prints the error.

But I don't think you should kill the application if you don't configure AWS correctly (you should log to help). Maybe a good strategy would be to have a refresh trigger to get new resources periodically.

Reply:
My point is this, suppose you call it

`honeyalb aaabc21wqw`

Where aaabc21wqw is the alb name, and alb _does **not** have logging configured_.

_Before_ your patch, it would exit.
_After_ your patch, the program will run forever, but it will do nothing (but print an error message).

You have changed the behavior a lot. The _user_ expects to get logs specifically from `aaabc21wqw`, but not only you are not delievering, instead of crashing/failing fast, you keep running.

What I have done with `honeyelb` and `honeycloudfront` is to _only ignore errors **if** no argument is supplied_, i.e., if you call the binary like this:

`honeyelb`
`honeycloudfront`

In this mode, the binary will look for _all_ the load balancers/cloudfront distros in the user account and read all/any logs enabled. So it makes sense that it would not fail if one of those is not enabled.

Reply:
> But I don't think you should kill the application if you don't configure AWS correctly (you should log to help).

I think so _if you list the load balancers by name_. If not, then you should try to get all resources in the account, and ignore errors.

>  Maybe a good strategy would be to have a refresh trigger to get new resources periodically.

That's a great idea! (but I think this would be a new feature altogether, and unrelated to the previous points)

Reply:
> That's a great idea! (but I think this would be a new feature altogether, and unrelated to the previous points)

Agree

> You have changed the behavior a lot. The user expects to get logs specifically from aaabc21wqw, but not only you are not delievering, instead of crashing/failing fast, you keep running.

I does make sense no to change the current behavior in bugfix. I'll update to keep that.